streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -790,23 +790,18 @@ private int putNodeGroupName(final String nodeName,
     }
 
     public synchronized ProcessorTopology build() {
-        return build((Integer) null);
+        final Set<String> nodeGroup = new HashSet<>();
+        for (final Set<String> value : nodeGroups().values()) {
+            nodeGroup.addAll(value);
+        }
+        nodeGroup.removeAll(globalNodeGroups());
+
+        initializeSubscription();
+        return build(nodeGroup);
     }
 
     public synchronized ProcessorTopology build(final Integer topicGroupId) {
-        final Set<String> nodeGroup;
-        if (topicGroupId != null) {
-            nodeGroup = nodeGroups().get(topicGroupId);
-        } else {
-            // when topicGroupId is null, we build the full topology minus the global groups
-            final Set<String> globalNodeGroups = globalNodeGroups();
-            final Collection<Set<String>> values = nodeGroups().values();
-            nodeGroup = new HashSet<>();
-            for (final Set<String> value : values) {
-                nodeGroup.addAll(value);
-            }
-            nodeGroup.removeAll(globalNodeGroups);
-        }
+        final Set<String> nodeGroup = nodeGroups().get(topicGroupId);
```

Comment:
Should we check that topicGroupId is not null?

Reply:
I moved all calls  where `topicGroupId` would be null to now call the parameterless `build()` instead. We can actually make `topicGroupId` just a regular `int` now

Reply:
Sounds good.